File: plugin/scratch/ScratchBlock.js
// ScratchBlockPrototype.js
// Zilch Editor JavaScript Plugin API

import ScratchBlockPrototype from 'ScratchBlockPrototype'

'use strict';

class ScratchBlock {
    /**
     * Constructor
     * @param {ScratchBlockPrototype} blockprototype Prototype of this block
     * @param {ScratchBlock} lastblock Last block
     * @param {ScratchBlock} nextblock Next block
     */
    constructor(blockprototype, lastblock = null, nextblock = null) {
        this.blockprototype = blockprototype;
        this.lastblock = lastblock;
        this.nextblock = nextblock;
        this.x = 0;
        this.y = 0;
    }
}

export default ScratchBlock

File: editor/scratch/ScratchBlock.cpp
#include "ScratchBlock.h"
#include "../widget/BlockItem.h"

ScratchBlock::ScratchBlock(ScratchBlockPrototype *Block, ScratchBlock *LastBlock, ScratchBlock *NextBlock) {
	this->Block = Block;
	this->LastBlock = LastBlock;
	this->NextBlock = NextBlock;
	this->Item = nullptr;
	this->X = this->Y = 0;
}

ScratchBlock::~ScratchBlock() {
	delete this->Item;
}

bool ScratchBlock::isHead() {
	return this->LastBlock == nullptr;
}
